scc/src/opts.rs

```diff
@@ -50,85 +50,189 @@ pub fn fix_args(args: Vec<String>) -> Vec<String> {
+        .adjacent()
+        .anywhere()
+        .map(|pair| pair.1)
+        .optional()
+        .catch()
 }
 
 fn script_paths() -> impl Parser<Vec<PathBuf>> {
     let tag = any::<String>("-compile").guard(|s| s == "-compile", "not compile");
-    let value = positional::<PathBuf>("SCRIPT_PATH");
-    construct!(tag, value).anywhere().map(|pair| pair.1).many().catch()
+    let value = positional::<PathBuf>("SCRIPT_PATH")
+        .guard(|s| !s.to_str().unwrap().to_string().starts_with('-'), "starts with -");
+    construct!(tag, value)
+        .adjacent()
+        .anywhere()
+        .map(|pair| pair.1)
+        .many()
+        .catch()
 }
 
 fn script_paths_file() -> impl Parser<Option<PathBuf>> {
     let tag = any::<String>("-compilePathsFile")
-        .help("File listing all of the script paths")
+        .help("File containing a newline-delimiter list of redscript paths to compile")
         .guard(|s| s == "-compilePathsFile", "not compilePathsFile");
-    let value = positional::<PathBuf>("FILE");
-    construct!(tag, value).anywhere().map(|pair| pair.1).optional().catch()
+    let value = positional::<PathBuf>("SCRIPTS_PATHS_FILE")
+        .guard(|s| !s.to_str().unwrap().to_string().starts_with('-'), "starts with -");
+    construct!(tag, value)
+        .adjacent()
+        .anywhere()
+        .map(|pair| pair.1)
+        .optional()
+        .catch()
 }
 
 fn cache_dir() -> impl Parser<Option<PathBuf>> {
-    let tag = any::<String>("-customCacheDir").guard(|s| s == "-customCacheDir", "not customCacheDir");
-    let value = positional::<PathBuf>("CACHE_DIR");
-    construct!(tag, value).map(|pair| pair.1).optional().catch()
+    let tag = any::<String>("-customCacheDir")
+        .help("A custom cache dir to write the final.redscripts to")
+        .guard(|s| s == "-customCacheDir", "not customCacheDir");
+    let value = positional::<PathBuf>("CACHE_DIR")
+        .guard(|s| !s.to_str().unwrap().to_string().starts_with('-'), "starts with -");
+    construct!(tag, value)
+        .adjacent()
+        .anywhere()
+        .map(|pair| pair.1)
+        .optional()
+        .catch()
 }
 
 impl Opts {
-    pub fn load(args: &[&str]) -> Self {
-        let optimize = toggle_options("optimize", "Optimize the redscripts");
-        let threads = slong("-threads", "Number of theads to script_paths on")
-            .parse(|s| s.parse::<u8>())
-            .fallback(1);
-        let no_warnings = toggle_options("Wnone", "No warnings");
-        let no_testonly = toggle_options("no-testonly", "No testonly classes");
-        let no_breakpoint = toggle_options("no-breakpoint", "No breakpoints");
-        let profile_off = toggle_options("profile=off", "Profile off");
-        let cache_file = any::<PathBuf>("CACHE_FILE").optional();
+    pub fn get_parser() -> OptionParser<Opts> {
+        let optimize = toggle_options("-optimize", "Optimize")
+            .parse(|s| Ok::<bool, String>(s.unwrap_or(Opts::default().optimize)));
```

Comment:
can we extract the defaults to individual `const` items somewhere? and then use them both in the `default` impl and here

Reply:
also most of these don't need to use `parse` because they can't fail:
```suggestion
            .map(|s| s.unwrap_or(Opts::default().optimize));
```

Reply:
re `const`, would [something like this](https://stackoverflow.com/a/61892366/1530201) work, or would you prefer it separate? tbf, i don't think we really need `Default` itself - it could just be another static member on `Opts`.

Reply:
I just mean something like this:
```rs
const DEFAULT_OPTIMIZE: bool = false;
...
```rs
...
.map(|s| s.unwrap_or(DEFAULT_OPTIMIZE));
```
```rs
impl Default for Opts {
  fn default() -> Self {
    Self {
      ...
      optimize: DEFAULT_OPTIMIZE
    }
  }
}

Reply:
i guess just this would work:
```rust
impl Opts {
    pub const fn default() -> Self {
        Self {  
            script_paths: vec![],
            cache_dir: None,
            optimize: false,
            threads: 1,
            warnings: None,
            no_testonly: false,
            no_breakpoint: false,
            profile: false,
            script_paths_file: None,
            cache_file: None,
        }
    }
}
```

Reply:
it might make sense to get rid of the default impls and just use a few `const` items, I don't like creating all the default opts just to get a single value